### ISSUE TITLE
Please let me help you write this maunal

### ISSUE THREAD

**Issue by Teslarduino** [Tenkir/vanmoof-s2-user-manual#1]
Hi, i have managed to dissasemble the x2 and reverse engineered the bms, the charger cartridge, and the smart cartridge pinout, I would love to help you and write this manual with you!  I believe you are doing amazing work!

**Comment by Tenkir** [comment]
That would be awesome! I've got a few pages I've been meaning to write, but haven't gotten around to. Feel free to open a PR into this repo!

**Comment by Tenkir** [comment]
@Teslarduino Do you have the pinout for the smart catridge handy?

**Comment by Silly105** [comment]
Are you two still interested? DIYing Vanmoof bikes got quite a push since the bankruptcy. If you, @Teslarduino managed to reverse engineer the bms communication - right now there are quite a few people out there trying exactly that (and failing). Your help would be much appreciated.

**Comment by SeAlwin** [comment]
@Silly105 Since you have disassembled your S2.

Mij boost cable is broken very close to the steering wheel. Too close to strip, so I need to replace the very thin cable going into the bike.

My guess is that I need to access it through the front light (still need to get me a T10H to take out the light).

Do you known, or can check, how I can replace extend or replace that tiny cable coming out of the bike?
I assume this cable is connected to the internal connector, connecting the cartridge.

**Comment by Silly105** [comment]
@SeAlwin I did not take out the front lights. But there is no other connection between the bike and the smart cartridge, so the boost cable can't be anywhere else than on that connector,
If you like to discuss this further, please open an issue to https://github.com/Silly105/vanmoof-s2-user-manual/ so we don't spam this repo.

**Comment by Tenkir** [comment]
> @Silly105 Since you have disassembled your S2.
> 
> Mij boost cable is broken very close to the steering wheel. Too close to strip, so I need to replace the very thin cable going into the bike.
> 
> My guess is that I need to access it through the front light (still need to get me a T10H to take out the light).
> 
> Do you known, or can check, how I can replace extend or replace that tiny cable coming out of the bike? I assume this cable is connected to the internal connector, connecting the cartridge.

I've taken mine apart entirely. Have been planning to update this guide with schematics for the smart cartridge. Unfortunately the nooster wires are potted in epoxy so there's not a good way to replace them if they're damaged close to the connector.

All the wires attach to a harness. If there's enough cable left, you could remove the harness and solder on extensions.

Here's what the harness looks like:

![image](https://github.com/Tenkir/vanmoof-s2-user-manual/assets/3649828/da8ca8b3-aedf-4f2f-af55-6e37ce5df0e8)

**Comment by Silly105** [comment]
Is that a recent photo? Since I would like to work on documenting the battery controller, it would be great if you could provide a pinout plan for the battery connector (so which pin is which, referencing to your connection plan on the wiring harness page). A quick and dirty picture of a pen on paper sketch will suffice, I can create a presentable image from that.

**Comment by Tenkir** [comment]
> Is that a recent photo? Since I would like to work on documenting the battery controller, it would be great if you could provide a pinout plan for the battery connector (so which pin is which, referencing to your connection plan on the wiring harness page). A quick and dirty picture of a pen on paper sketch will suffice, I can create a presentable image from that.

Yes, I'll upload the schematics when I'm back in the studio later this week.

**Comment by Tenkir** [comment]
@Silly105 it looks like you added the pins on the battery side in #2. Do you still need these pinouts?

I'm working on analyzing the logic for the battery communication protocol, as I'm again encountering the Error 29 communication error. Have you made any efforts on this previously?

**Comment by Silly105** [comment]
I just tried to reverse engineer the battery, what didn't do is the smart cartridge side. So if you know the wiring from battery connector to smart cartridge connector that would be very helpful, i think.

About the comm protocol... I have not done any actual testing, but a guy on Reddit found strong hints that it's a rs232 protocol. Maybe that helps.

**Comment by Tenkir** [comment]
![image (1)](https://github.com/Tenkir/vanmoof-s2-user-manual/assets/3649828/64f6a20a-ee07-4aab-ae5a-4a54d24cfc4f)

This is the pinout that corresponds to the schematic in [Wiring Harness](https://github.com/Tenkir/vanmoof-s2-user-manual/blob/main/wiringHarness.md) section. Is this what you where looking for? That lines up with your annotations [here](https://github.com/Tenkir/vanmoof-s2-user-manual/blob/main/battery.md#pinout).

**Comment by Tenkir** [comment]
I picked up two new batteries on ebay yesturday. Interestingly, the seller had this photo of checking pack voltage using the communication pinout. Looks like D3 is a + voltage pin for communicating pack voltage (assuming the markings on the battery indicating polarity are correct).

![s-l1600](https://github.com/Tenkir/vanmoof-s2-user-manual/assets/3649828/d1046cff-1d48-41cc-9128-3d0b5f30a5ed)
